src/SignalR/server/Core/src/HubConnectionContext.cs

```diff
@@ -146,6 +147,7 @@ internal StreamTracker StreamTracker
         // Currently used only for streaming methods
         internal ConcurrentDictionary<string, CancellationTokenSource> ActiveRequestCancellationSources { get; } = new ConcurrentDictionary<string, CancellationTokenSource>(StringComparer.Ordinal);
 
+        [SuppressMessage("ApiDesign", "RS0026:Do not add multiple overloads with optional parameters", Justification = "Required to maintain compatibility")]
```

Comment:
nit: Should we craft a better `Justification` for the many `T1 parameter, CancellationToken token = default`, `T2 parameter, CancellationToken token = default`, &hellip; cases❔ If we were adding `HubConnectionContext` tomorrow, I bet we'd use the same set of overloads. Perhaps,
```suggestion
        [SuppressMessage("ApiDesign", "RS0026:Do not add multiple overloads with optional parameters", Justification = "May need a CancellationToken with either of these message types.")]
```
Of course, if you do that here, we'd need to look at the rest of our open PRs and I'm not sure it's worthwhile.